File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        int a = nums[0];
        int b = nums[n-1];
        ArrayList<Integer> arr = new ArrayList<>();
        HashSet<Integer> set = new HashSet<>();
        for(int i : nums){
            set.add(i);
            
        }
        for(int i=a; i<b; i++){
            if (!set.contains(i)){
                arr.add(i);
            }    
        }
        return arr;
    }
}

File: 3731-find-missing-elements/3731-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        n = len(nums)
        nums.sort()
        a = nums[0]
        b = nums[n-1]
        
        missing = []
        
        for i in range(a,b+1):
            if i not in set(nums):
                missing.append(i)
        return missing        

